tests/core/numerics/interpolator/interpolator_test.py

```diff
@@ -43,5 +43,15 @@ def main():
 
 
 
+def interpolate_datasets(beforeTime, afterTime, interp_time, beforeData, afterData):
+    alpha = (interp_time - beforeTime) / (afterTime - beforeTime)
```

Comment:
you probably want to assert afterTime > beforeTime and that interp_time is between the two, otherwise you'll get invalid alpha (can't interpolate outside time limits)

Reply:
```
    assert before_time < after_time
    assert before_time <= interp_time <= after_time
```

"before_time" must be less than "after_time"
but "interp_time" can be equal to either, or between

Reply:
interp_time can be equal to before and after times, that means alpha 0 or 1, respectively, indeed. However checking equality with floats is tricky so you want to check if it's not < before and not > after, I guess

Reply:
how bout something like this 
![image](https://user-images.githubusercontent.com/30491/112453470-07a7ad80-8d58-11eb-9345-50330e266cf4.png)